### DISCUSSION TITLE
VSCode: Going to definitions withing Java code

### DISCUSSION THREAD

**Discussion by hmf** [scalameta/metals#2382] [Q&A] [answered]
When using the VSCode IDE I can go to a definition from Scala to Java. However, once in the Java code, the go-to definition functionality does not seem to work. 

Is it wishful thinking to ask for this for future versions? An maybe even finding the imports of undefined symbols (quick fix)?

TIA

**Comment by gabro** [comment, marked as answer]
@hmf thanks for asking! Metals is a Scala Language Server and it will likely stay that way.

That said, the Java Language Server exists and it's excellent, so ideally you should use Metals when working on Scala files and the Java LS when working on Java files.

For that to work seamlessly, we need to solve this https://github.com/scalameta/metals-feature-requests/issues/5

**Reply by hmf** [reply]
Ahhh, missed that. Keeping tabs on the issue. 

Thank you.
